File: src/main/java/com/kubemetrics/user/UserService.java
package com.kubemetrics.user;

import org.springframework.stereotype.Service;

import java.util.Optional;

@Service
class UserService {
    private final UserRepository userRepository;

    public UserService(UserRepository userRepository) {
        this.userRepository = userRepository;
    }

    public User createUser(User user) {
        return userRepository.save(user);
    }

    public Optional<User> findUserById(String id) {
        return userRepository.findById(id);
    }
}

File: src/main/kotlin/com/kubemetrics/user/UserService.kt
package com.kubemetrics.user

import org.springframework.data.repository.CrudRepository
import org.springframework.stereotype.Service
import java.util.Optional


@Service
class UserService(val userRepository: UserRepository) {
    fun createUser(user: User): User {
        return userRepository.save(user)
    }

    fun findUserById(id: String): Optional<User> {
        return userRepository.findById(id)
    }
}

interface UserRepository : CrudRepository<User, String>

data class User(val id: String? = null,
                val firstName: String,
                val lastName: String,
                val email: String)

